dotnet/src/Agents/Abstractions/AgentThread.cs

```diff
@@ -45,6 +92,10 @@ protected internal virtual async Task CreateAsync(CancellationToken cancellation
         }
 
         this.Id = await this.CreateInternalAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+
+#pragma warning disable SKEXP0110 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+        await this.AIContextBehaviors.OnThreadCreatedAsync(this.Id!, cancellationToken).ConfigureAwait(false);
```

Comment:
I'm not sure the IDE will warn after a `is not null` check. (same for others)

```suggestion
        await this.AIContextBehaviors.OnThreadCreatedAsync(this.Id, cancellationToken).ConfigureAwait(false);
```

Reply:
Thanks, fixed here and below.